### DISCUSSION TITLE
Using PyInstaller with Python installed from PDM

### DISCUSSION THREAD

**Discussion by JimiC** [pdm-project/pdm#3209] [Ideas]
TL;DR Installing a `Python` version with `PDM` doesn't play nice with `PyInstaller` and the cause is 'probably' missing `lib-dynload` files.

Recently I discovered `PDM` and got exited on using it. I use `PyInstaller` on many of my projects as I like the idea of using executables to run my apps. 
Note: With the previous tool I used, everything worked fine.

So, I switched one of my projects to `PDM` only to find out that the executable from `PyInstaller` was not working anymore, complaining about missing `Python` modules.
To keep it short, upon debugging I noticed that `PyInstaller` couldn't find files from `lib-dynload`, so it didn't copied the said folder into the package. Inspecting the folder from the `Python` version, that `PDM` installed, I found only 2 files, which where irrelevant to my case.
That's why `PyInstaller` ignored it!!!

Workaround: Installing `PyEnv` and using it as the `Python` version manager, instead of `PDM`, resolved the issue and the executable started to work again.

Proposed solutions: 
  - Notify `indygreg` about the case and have his `Python` binaries include the proper `lib-dynload` files.
  - Give the ability to specify the source of the `Python` builds (could be `pyenv`'s source)
  - Give the ability to install a `Python` version from source
  - Something else
  
  Some or all the above proposed solutions may already be available but as a new user I probably haven't discovered them yet.
  
  Anyways, thanks @frostming for creating an awesome tool.
  
 * Reference: https://github.com/indygreg/python-build-standalone/discussions/369

**Comment by frostming** [comment]
`pdm python` doesn't handle installation source other than inygreg's. You can install Python yourself with whatever tool you like and make PDM able to find the installations, pyenv location is supported by default.

**Reply by JimiC** [reply]
Fair enough.
The idea is to be able to remove `pyenv` dependency entirely.
I should probably take this up to inygreg then.
